src/Forms/FormMessage.php

```diff
@@ -92,6 +91,57 @@ public function setMessage(
         return $this;
     }
 
+    /**
+     * Appends a message to the existing message if the types match, overrides if not
+     *
+     * @param string $message Message string
+     * @param string $messageType Message type
+     * @param string $messageCast
+     * @param bool $force Whether or not to override the existing message if appending is invalid
+     * @throws InvalidArgumentException
+     * @return $this
+     */
+    public function appendMessage(
+        string $message,
+        string $messageType = ValidationResult::TYPE_ERROR,
+        string $messageCast = ValidationResult::CAST_TEXT,
+        bool $force = false,
+    ): static {
+        if (empty($this->message)) {
+            return $this->setMessage($message, $messageType, $messageCast);
+        }
+        $canBeMerged = ($messageType === $this->getMessageType() && $messageCast === $this->getMessageCast());
+
+        if (!$canBeMerged && $force === false) {
+            throw new InvalidArgumentException(
+                sprintf(
+                    "Couldn't append message of type %s and cast %s to existing message of type %s and cast %s",
+                    $messageType,
+                    $messageCast,
+                    $this->getMessageType(),
+                    $this->getMessageCast(),
+                )
+            );
+        }
+
+        // Checks that the exact message string is not already contained before appending
+        $messageContainsString = strpos($this->message, $message) !== false;
+        if ($canBeMerged && $messageContainsString) {
+            return $this;
+        }
```

Comment:
Can you please explain why you want to add this? It could potentially result in unexpectedly not including messages that the developer wants to include if an existing message has it as a part of its message.

Reply:
In testing I was seeing duplicate messages when entering a password that didn't meet the criteria. I didn't look into it _too_ much, but my general feeling was that without a major refactor (a la #10967) I couldn't effectively prevent duplicates. (actually thinking about this while I'm typing, I might "just" be able to use `array_unique()` on a private property 🤔  - will try that for a bit)

My theory (didn't go too deep) was that more than one thing was calling `validate()` on the `Member` record on `save()` - my guess would be the form handler and in `onBeforeWrite()`. That probably feels like a thing we could fix too but I didn't want to go too deep down the rabbit hole.

Filtering for unique messages in `Form::loadMessagesFrom()` didn't help because it was actually two separate calls.

Reply:
Ahh yup that's fair enough.
I'd say don't spend too much time trying to find a cleaner solution - I'd be satisfied with a note in the PHPDoc about this limitation (so at least it's documented) and it can be solved more fully in a future major release (#10967).